compiler-core/src/language_server/server.rs

```diff
@@ -181,12 +205,57 @@ where
                 self.watched_files_changed(params)
             }
 
+            "workspace/didChangeConfiguration" => {
+                self.request_configuration();
+                Feedback::default()
+            }
+
             _ => return,
         };
 
         self.publish_feedback(feedback);
     }
 
+    fn handle_response(&mut self, handler: ResponseHandler, result: Json) {
+        match handler {
+            ResponseHandler::UpdateConfiguration => {
+                match serde_json::from_value::<Vec<Configuration>>(result) {
+                    Ok(result) => {
+                        if let Some(config) = result.into_iter().next() {
+                            self.config = VersionedConfig {
+                                version: self.config.version + 1,
+                                config,
+                            };
+                        }
+                    }
+                    Err(err) => eprintln!("unable to parse configuration: {err}"),
```

Comment:
This silently discards the problem from the user's point of view, we should surface it to them somehow. Crashing would be the easiest way, but an actual error message would be better. I'm happy with either.